src/test/scala/mesosphere/marathon/api/v2/json/V2AppDefinitionTest.scala

```diff
@@ -49,11 +49,13 @@ class V2AppDefinitionTest extends MarathonSpec with Matchers {
     shouldViolate(app, "id", idError)
 
     app = app.copy(id = "-dash-disallowed-at-start".toRootPath)
-    validateJsonSchema(app, false)
+    // The JSON schema does allow this. It is not as strict as our internal validation to keep the regular
+    // expression reasonably simple.
```

Comment:
I think you can use the regex from the cli on group ids. It is more complex though... See [here](https://github.com/mesosphere/dcos-cli/blob/master/cli/dcoscli/data/marathon-group-schema.json#L210)